spec/helpers.lua

```diff
@@ -3662,27 +3638,26 @@ end
       pid_path = running_conf.nginx_pid
     end
 
-    return kill.kill(pid_path, signal)
+    return process.signal(pid_path, signal)
   end,
-  -- send signal to all Nginx workers, not including the master
-  signal_workers = function(prefix, signal, pid_path)
-    if not pid_path then
-      local running_conf = get_running_conf(prefix)
-      if not running_conf then
-        error("no config file found at prefix: " .. prefix)
-      end
 
-      pid_path = running_conf.nginx_pid
+  -- send signal to all Nginx workers, not including the master
+  signal_workers = function(signal)
+    local workers = get_kong_workers()
+    for _, pid in ipairs(workers) do
+      process.signal(pid, signal)
     end
 
-    local cmd = string.format("pkill %s -P `cat %s`", signal, pid_path)
-    local _, code = pl_utils.execute(cmd)
-
-    if not pid_dead(pid_path) then
-      return false
-    end
+    wait_until(function()
```

Comment:
_before_ this change, this function was calling `pid_dead("/path/to/nginx.pid")`, which had some problems:

1. it doesn't make sense to check the master pid given we just wanted to kill worker processes
2. `pid_dead` is supposed to take a pid as input (not a pid file), so it would blindly call `ps -p /path/to/nginx.pid`. This returns `1` due to it being improper usage of `ps`, causing `pid_dead` to think "okay cool, the process has already exited, so I'm done".